### PR TITLE
#61 WORKING WITH ABSTRACTION COMMIT

### PR DESCRIPTION
**Pull request by ivelin980**

### CODE REVIEW

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstracion/lab/cards/with/power/Card.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package jv.oop.working.with.abstracion.lab.cards.with.power;
+
+public enum Card {
+	ACE(14), TWO(2), THREE(3), FOUR(4), FIVE(5), SIX(6), SEVEN(7), EIGHT(8), NINE(9), TEN(10), JACK(11), QUEEN(12),
+	KING(13);
+	
+	private int power;
+	
+	Card(int power) {
+	this.power = power;
```
Comment (ngavrailov) [inline javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstracion/lab/cards/with/power/Card.java:10]: formatting

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/exercise/jedi/galaxy/Main.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			}
+
+			jediCol++;
+			jediRow--;
+		}
+		return sum;
+	}
+
+	private static int[] getCoordinates(String command) {
+		int[] playerCoordinates = Arrays.stream(command.split(" ")).mapToInt(Integer::parseInt).toArray();
```
Comment (ngavrailov) [inline javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/exercise/jedi/galaxy/Main.java:70]: You don't need this assign. You could just return it immediately.

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/PriceCalculator.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	
+	public PriceCalculator(double price, int days, Seasons season, DiscountType discount) {
+		this.price = price;
+		this.days = days;
+		this.season = season;
+		this.discount = discount;
+	}
+	
+	public double calculatePrice() {
+		return this.price*this.days*this.season.getMultiplier()*(1.00-(this.discount.getDiscount()/100.00));
```
Comment (ngavrailov) [inline javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/PriceCalculator.java:17]: formatting

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/Seasons.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+public enum Seasons {
+	AUTUMN(1),
+	SPRING(2),
+	WINTER(3),
+	SUMMER(4);
+	
+	private int multiplier;
+	
+	Seasons(int multiplier) {
+	this.multiplier = multiplier;
```
Comment (ngavrailov) [inline javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/Seasons.java:12]: formatting

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/rhombus/stars/RhombusOfStars.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+public class RhombusOfStars {
+
+	public static void main(String[] args) {
+		Scanner scan = new Scanner(System.in);
+		int n = Integer.parseInt(scan.nextLine());
+		printRhombus(n);
+		scan.close();
+	}
+
+	private static void printRhombus(int rows) {
```
Comment (ngavrailov) [inline javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/rhombus/stars/RhombusOfStars.java:22]: Good job to separate the parts of the method in different methods.

**Review by ngavrailov — CHANGES_REQUESTED** [review]

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/rhombus/stars/RhombusOfStars.java (hunk 5 above)
Comment (ivelin980) [reply javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/rhombus/stars/RhombusOfStars.java:22]: thank you

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/Seasons.java (hunk 4 above)
Comment (ivelin980) [reply javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/Seasons.java:12]: fixed

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/PriceCalculator.java (hunk 3 above)
Comment (ivelin980) [reply javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/lab/hotel/reservation/PriceCalculator.java:17]: fixed

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/exercise/jedi/galaxy/Main.java (hunk 2 above)
Comment (ivelin980) [reply javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstraction/exercise/jedi/galaxy/Main.java:70]: fixed

File: javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstracion/lab/cards/with/power/Card.java (hunk 1 above)
Comment (ivelin980) [reply javaOOP_workingWithAbstraction/src/jv/oop/working/with/abstracion/lab/cards/with/power/Card.java:10]: fixed

**Review by ngavrailov — APPROVED** [review]

**Pull request merged**